src/librustc/mir/tcx.rs

```diff
@@ -124,20 +118,19 @@ impl<'a, 'gcx, 'tcx> PlaceTy<'tcx> {
                     }
                 }
             }
-            ProjectionElem::Downcast(adt_def1, index) =>
-                match self.to_ty(tcx).sty {
-                    ty::Adt(adt_def, substs) => {
+            ProjectionElem::Downcast(_name, index) => {
+                let ty = self.to_ty();
+                match &ty.sty {
+                    ty::Adt(adt_def, _substs) => {
```

Comment:
You can remove this `match`, it's kind of pointless nowadays.

Reply:
done